webrecorder/browsertrix
# Allow users to set max depth/hop out within scope

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/701



### Changes

Adds an input to the Workflow creation and edit form for specifying crawl depth. This input is conditionally shown for seeded crawls when the scope is set to "Pages on this domain", "Pages on this domain & subdomains" or "Custom page prefix". The "any" scope is also supported for backwards compatibility but is not shown by default or in new configs.

API implementation: The `depth` value is set in the primary seed config, i.e. the first seed in `seeds: []`, not in the outer `.config.depth` property.

### Manual testing

1. Log in and go to Crawling -> New Crawl Workflow -> Seeded Crawl
2. Choose "Pages on this domain" from the Start URL Scope dropdown. Verify Max Depth field is visible. Repeat for "Pages on this domain & subdomains" and "Custom page prefix"
3. Enter a Max Depth value and go to Review Config. Verify value is shown correctly under Crawler Settings
4. Save. Verify value is shown correctly on Workflow detail page
5. Click Actions -> Edit Workflow Settings. Verify you're able to remove, update, and otherwise save the Max Depth as expected.

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| New Workflow - Scope tab | <img width="868" alt="Screen Shot 2023-05-03 at 12 01 03 PM" src="https://user-images.githubusercontent.com/4672952/236026196-06ac21f9-a276-48e9-8098-edcb94076c92.png"> |
| Workflow Detail | **Value specified:**<img width="88" alt="Screen Shot 2023-05-03 at 12 24 55 PM" src="https://user-images.githubusercontent.com/4672952/236026282-f27ce5da-9fb9-47a6-bf31-bfa027d2b7cf.png"></br>**Value not specified:**<img width="95" alt="Screen Shot 2023-05-03 at 12 25 35 PM" src="https://user-images.githubusercontent.com/4672952/236026417-2cbae3b5-2df3-43e1-be1e-ee343854bcc5.png"> |
